include/swift/Runtime/Config.h

```diff
@@ -38,7 +38,7 @@
 #ifndef SWIFT_USE_SWIFTCALL
 // Clang doesn't support mangling functions with the swiftcall attribute
 // on Windows and crashes during compilation: http://bugs.llvm.org/show_bug.cgi?id=32000
-#if (__has_attribute(swiftcall) || defined(__linux__)) && !defined(_WIN32)
+#if __has_attribute(swiftcall) && !defined(_WIN32)
```

Comment:
Yes, I think System.h is probably a reasonable place to do this.

TableGen would be a nice way of doing this because you can do a lot of nice inheritance things with TableGen classes; e.g. you can make the Darwin x86-64 target inherit from a generic x86-64 target and, I dunno, use a more aggressive invalid-pointer set or turn on swiftcc or something.  It would also make it really easy to make sure that all the settings are set for all the targets; you'll notice that System.h basically requires clients to know what settings are actually set there.  It would also make it really easy to make the same information available in different ways to different clients; e.g. you should really only have to say "yes, use swiftcc" once, but IRGen probably wants to know that as an llvm::CallingConv::ID and Runtime probably wants to know that as a macro that expands to a calling-convention attribute.

Things that I think would make sense to put in the target configuration
  - everything from System.h
  - maybe everything from SwiftTargetInfo.h
  - whether the target uses swiftcc
  - whether the target uses the preserve-most convention